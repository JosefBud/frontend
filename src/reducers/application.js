import { buildActions } from 'utils';

export const types = buildActions('application', [
  'INIT_APP',
  'LOGIN_USER',
  'POP_TOAST',
  'ADD_TOAST',
  'REMOVE_TOAST',
  'HIDE_TOAST',
  'REQUEST_TOKEN',
  'REQUEST_TOKEN_SUCCESS',
  'REQUEST_TOKEN_FAILURE',
  'REQUEST_CURRENT_USER',
  'REQUEST_CURRENT_USER_SUCCESS',
  'REQUEST_CURRENT_USER_FAILURE',
  'LOGIN_USER_SUCCESS',
  'LOGIN_USER_FAILURE',
  'LOGOUT_USER',
  'LOGOUT_USER_SUCCESS',
  'LOGOUT_USER_FAILURE',
  'REGISTER_USER',
  'REGISTER_USER_SUCCESS',
  'REGISTER_USER_FAILURE',
  'REQUEST_USERS',
  'REQUEST_USERS_SUCCESS',
  'REQUEST_USERS_FAILURE',
  'REQUEST_MIGRATIONS',
  'REQUEST_MIGRATIONS_SUCCESS',
  'REQUEST_MIGRATIONS_FAILURE',
  'REQUEST_ROLES',
  'REQUEST_ROLES_SUCCESS',
  'REQUEST_ROLES_FAILURE',
  'REQUEST_FLAVORS',
  'REQUEST_FLAVORS_SUCCESS',
  'REQUEST_FLAVORS_FAILURE'
]);

const initApp = () => ({
  type: types.INIT_APP
});

const loginUser = (emailAddress, password) => ({
  type: types.LOGIN_USER,
  emailAddress,
  password
});

const popToast = toast => ({
  type: types.POP_TOAST,
  toast
});

const addToast = toast => ({
  type: types.ADD_TOAST,
  toast
});

const removeToast = id => ({
  type: types.REMOVE_TOAST,
  id
});

const hideToast = id => ({
  type: types.HIDE_TOAST,
  id
});

const requestToken = (emailAddress, password) => ({
  type: types.REQUEST_TOKEN,
  emailAddress,
  password
});

const requestTokenSuccess = ({ token, expiration }) => ({
  type: types.REQUEST_TOKEN_SUCCESS,
  expiration,
  token
});

const requestTokenFailure = error => ({
  type: types.REQUEST_TOKEN_FAILURE,
  error
});

const requestCurrentUser = () => ({
  type: types.REQUEST_CURRENT_USER
});

const requestCurrentUserSuccess = user => ({
  type: types.REQUEST_CURRENT_USER_SUCCESS,
  user
});

const requestCurrentUserFailure = error => ({
  type: types.REQUEST_CURRENT_USER_FAILURE,
  error
});

const loginUserSuccess = () => ({
  type: types.LOGIN_USER_SUCCESS
});

const loginUserFailure = error => ({
  type: types.LOGIN_USER_FAILURE,
  error
});

const logoutUser = () => ({
  type: types.LOGOUT_USER
});

const logoutUserSuccess = () => ({
  type: types.LOGOUT_USER_SUCCESS
});

const logoutUserFailure = error => ({
  type: types.LOGOUT_USER_FAILURE,
  error
});

const registerUser = details => ({
  type: types.REGISTER_USER,
  details
});

const registerUserSuccess = () => ({
  type: types.REGISTER_USER_SUCCESS
});

const registerUserFailure = error => ({
  type: types.REGISTER_USER_FAILURE,
  error
});

const requestUsers = () => ({
  type: types.REQUEST_USERS
});

const requestUsersSuccess = users => ({
  type: types.REQUEST_USERS_SUCCESS,
  users
});

const requestUsersFailure = error => ({
  type: types.REQUEST_USERS_FAILURE,
  error
});

const requestMigrations = () => ({
  type: types.REQUEST_MIGRATIONS
});

const requestMigrationsSuccess = migrations => ({
  type: types.REQUEST_MIGRATIONS_SUCCESS,
  migrations
});

const requestMigrationsFailure = error => ({
  type: types.REQUEST_MIGRATIONS_FAILURE,
  error
});

const requestRoles = () => ({
  type: types.REQUEST_ROLES
});

const requestRolesSuccess = roles => ({
  type: types.REQUEST_ROLES_SUCCESS,
  roles
});

const requestRolesFailure = error => ({
  type: types.REQUEST_ROLES_FAILURE,
  error
});

const requestFlavors = () => ({
  type: types.REQUEST_FLAVORS
});

const requestFlavorsSuccess = flavors => ({
  type: types.REQUEST_FLAVORS_SUCCESS,
  flavors
});

const requestFlavorsFailure = error => ({
  type: types.REQUEST_FLAVORS_FAILURE,
  error
});

export const actions = {
  initApp,
  loginUser,
  popToast,
  addToast,
  removeToast,
  hideToast,
  requestToken,
  requestTokenSuccess,
  requestTokenFailure,
  requestCurrentUser,
  requestCurrentUserSuccess,
  requestCurrentUserFailure,
  loginUserSuccess,
  loginUserFailure,
  logoutUser,
  logoutUserSuccess,
  logoutUserFailure,
  registerUser,
  registerUserSuccess,
  registerUserFailure,
  requestUsers,
  requestUsersSuccess,
  requestUsersFailure,
  requestMigrations,
  requestMigrationsSuccess,
  requestMigrationsFailure,
  requestRoles,
  requestRolesSuccess,
  requestRolesFailure,
  requestFlavors,
  requestFlavorsSuccess,
  requestFlavorsFailure
};

export const initialState = {
  loggingIn: false,
  loggingOut: false,
  user: null,
  error: null,
  authorization: {
    accessToken: null,
    refreshToken: null,
    expiration: null
  },
  registration: {
    registering: false,
    complete: false,
    error: null
  },
  toasts: [],
  users: [],
  migrations: [],
  roles: [],
  flavors: []
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.ADD_TOAST:
      return {
        ...state,
        toasts: [...state.toasts, action.toast]
      };
    case types.REMOVE_TOAST:
      return {
        ...state,
        toasts: state.toasts.filter(toast => toast.id !== action.id)
      };
    case types.HIDE_TOAST: {
      const originalToast = state.toasts.find(toast => toast.id === action.id);

      if (!originalToast) {
        return state;
      }

      const newToast = {
        ...originalToast,
        show: false
      };
      const filteredToasts = state.toasts.filter(
        toast => toast.id !== action.id
      );

      return {
        ...state,
        toasts: [...filteredToasts, newToast]
      };
    }
    case types.LOGIN_USER:
      return {
        ...state,
        loggingIn: true
      };
    case types.LOGOUT_USER:
      return {
        ...state,
        loggingOut: true
      };
    case types.REGISTER_USER:
      return {
        ...state,
        registration: {
          ...state.registration,
          registering: true,
          details: action.details
        }
      };
    case types.REQUEST_TOKEN_SUCCESS:
      return {
        ...state,
        authorization: {
          ...state.authorization,
          accessToken: action.token,
          expiration: action.expiration
        }
      };
    case types.REQUEST_TOKEN_FAILURE:
      return {
        ...state,
        authorization: {
          ...state.authorization,
          error: action.error
        }
      };
    case types.REQUEST_CURRENT_USER_SUCCESS:
      return {
        ...state,
        user: action.user
      };
    case types.LOGOUT_USER_SUCCESS:
      return {
        ...state,
        loggingOut: false,
        user: null,
        authorization: initialState.authorization
      };
    case types.LOGIN_USER_SUCCESS:
      return {
        ...state,
        loggingIn: false
      };
    case types.LOGIN_USER_FAILURE:
    case types.LOGOUT_USER_FAILURE:
      return {
        ...state,
        loggingIn: false,
        loggingOut: false,
        error: action.error
      };
    case types.REGISTER_USER_SUCCESS:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: null
        }
      };
    case types.REGISTER_USER_FAILURE:
      return {
        ...state,
        registration: {
          registering: false,
          complete: true,
          error: action.error
        }
      };
    case types.REQUEST_USERS_SUCCESS:
      return {
        ...state,
        users: action.users
      };
    case types.REQUEST_USERS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.REQUEST_MIGRATIONS_SUCCESS:
      return {
        ...state,
        migrations: action.migrations
      };
    case types.REQUEST_MIGRATIONS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.REQUEST_ROLES_SUCCESS:
      return {
        ...state,
        roles: action.roles
      };
    case types.REQUEST_ROLES_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.REQUEST_FLAVORS_SUCCESS:
      return {
        ...state,
        flavors: action.flavors
      };
    case types.REQUEST_FLAVORS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
