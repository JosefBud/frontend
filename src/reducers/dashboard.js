import { buildActions } from 'utils';

export const types = buildActions('dashboard', [
  'REQUEST_MIGRATIONS',
  'REQUEST_MIGRATIONS_SUCCESS',
  'REQUEST_MIGRATIONS_FAILURE'
]);

const requestMigrations = () => ({
  type: types.REQUEST_MIGRATIONS
});

const requestMigrationsSuccess = migrations => ({
  type: types.REQUEST_MIGRATIONS_SUCCESS,
  migrations
});

const requestMigrationsFailure = error => ({
  type: types.REQUEST_MIGRATIONS_FAILURE,
  error
});

export const actions = {
  requestMigrations,
  requestMigrationsSuccess,
  requestMigrationsFailure
};

export const initialState = {
  migrations: []
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_MIGRATIONS_SUCCESS:
      return {
        ...state,
        migrations: action.migrations
      };
    case types.REQUEST_MIGRATIONS_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
