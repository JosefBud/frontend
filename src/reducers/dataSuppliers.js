import { buildActions } from 'utils';

export const types = buildActions('dataSuppliers', [
  'REQUEST_DATA_SUPPLIER',
  'REQUEST_DATA_SUPPLIER_SUCCESS',
  'REQUEST_DATA_SUPPLIERS',
  'REQUEST_DATA_SUPPLIERS_SUCCESS',
  'REQUEST_FAILURE'
]);

const requestDataSupplier = dataSupplierId => ({
  type: types.REQUEST_DATA_SUPPLIER,
  dataSupplierId
});

const requestDataSupplierSuccess = dataSupplier => ({
  type: types.REQUEST_DATA_SUPPLIER_SUCCESS,
  dataSupplier
});

const requestDataSuppliers = pager => ({
  type: types.REQUEST_DATA_SUPPLIERS,
  pager
});

const requestDataSuppliersSuccess = (dataSuppliers, pager) => ({
  type: types.REQUEST_DATA_SUPPLIERS_SUCCESS,
  dataSuppliers,
  pager
});

const requestFailure = error => ({
  type: types.REQUEST_FAILURE,
  error
});

export const actions = {
  requestDataSupplier,
  requestDataSupplierSuccess,
  requestDataSuppliers,
  requestDataSuppliersSuccess,
  requestFailure
};

export const initialState = {
  dataSupplier: {},
  cache: [],
  collection: [],
  pager: {
    count: null,
    limit: 100,
    page: 1,
    pages: null
  }
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_DATA_SUPPLIER_SUCCESS:
      return {
        ...state,
        dataSupplier: action.dataSupplier
      };
    case types.REQUEST_DATA_SUPPLIERS_SUCCESS:
      return {
        ...state,
        cache: action.dataSuppliers,
        collection: action.dataSuppliers[action.pager.page],
        pager: {
          ...state.pager,
          ...action.pager
        }
      };
    case types.REQUEST_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
