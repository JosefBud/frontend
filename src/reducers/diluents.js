import { buildActions } from 'utils';

export const types = buildActions('diluents', [
  'REQUEST_DILUENT',
  'REQUEST_DILUENT_SUCCESS',
  'REQUEST_DILUENTS',
  'REQUEST_DILUENTS_SUCCESS',
  'REQUEST_FAILURE'
]);

const requestDiluent = diluentId => ({
  type: types.REQUEST_DILUENT,
  diluentId
});

const requestDiluentSuccess = diluent => ({
  type: types.REQUEST_DILUENT_SUCCESS,
  diluent
});

const requestDiluents = pager => ({
  type: types.REQUEST_DILUENTS,
  pager
});

const requestDiluentsSuccess = (diluents, pager) => ({
  type: types.REQUEST_DILUENTS_SUCCESS,
  diluents,
  pager
});

const requestFailure = error => ({
  type: types.REQUEST_FAILURE,
  error
});

export const actions = {
  requestDiluent,
  requestDiluentSuccess,
  requestDiluents,
  requestDiluentsSuccess,
  requestFailure
};

export const initialState = {
  diluent: {},
  cache: [],
  collection: [],
  pager: {
    count: null,
    limit: 100,
    page: 1,
    pages: null
  }
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_DILUENT_SUCCESS:
      return {
        ...state,
        diluent: action.diluent
      };
    case types.REQUEST_DILUENTS_SUCCESS:
      return {
        ...state,
        cache: action.diluents,
        collection: action.diluents[action.pager.page],
        pager: {
          ...state.pager,
          ...action.pager
        }
      };
    case types.REQUEST_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
