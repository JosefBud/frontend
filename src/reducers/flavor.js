import { buildActions } from 'utils';

export const types = buildActions('flavor', [
  'REQUEST_STASH',
  'REQUEST_STASH_SUCCESS',
  'REQUEST_STASH_FAILURE',
  'ADD_STASH',
  'ADD_STASH_SUCCESS',
  'ADD_STASH_FAILURE'
]);

const requestStash = () => ({
  type: types.REQUEST_STASH
});

const requestStashSuccess = stash => ({
  type: types.REQUEST_STASH_SUCCESS,
  stash
});

const requestStashFailure = error => ({
  type: types.REQUEST_STASH_FAILURE,
  error
});

const addStash = flavor => ({
  type: types.ADD_STASH,
  flavor
});

const addStashSuccess = () => ({
  type: types.ADD_STASH_SUCCESS
});

const addStashFailure = error => ({
  type: types.ADD_STASH_FAILURE,
  error
});

export const actions = {
  requestStash,
  requestStashSuccess,
  requestStashFailure,
  addStash,
  addStashSuccess,
  addStashFailure
};

export const initialState = {
  loaded: false,
  error: null,
  stash: []
};

export const reducer = (state = initialState, action = {}) => {
  switch (action.type) {
    case types.REQUEST_STASH_SUCCESS:
      return {
        ...state,
        stash: action.stash,
        loaded: true
      };
    case types.REQUEST_STASH_FAILURE:
      return {
        ...state,
        error: action.error
      };
    case types.ADD_STASH_SUCCESS:
      return {
        ...state,
        loaded: false
      };
    case types.ADD_STASH_FAILURE:
      return {
        ...state,
        error: action.error
      };
    default:
      return state;
  }
};
