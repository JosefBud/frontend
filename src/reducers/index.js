import { combineReducers } from 'redux';
import {
  reducer as application,
  initialState as applicationState
} from './application';
import {
  reducer as dashboard,
  initialState as dashboardState
} from './dashboard';
import {
  reducer as dataSuppliers,
  initialState as dataSuppliersState
} from './dataSuppliers';
import { reducer as flavors, initialState as flavorsState } from './flavors';
import { reducer as roles, initialState as rolesState } from './roles';
import { reducer as toast, initialState as toastState } from './toast';
import { reducer as users, initialState as usersState } from './users';

export const initialState = {
  application: applicationState,
  dashboard: dashboardState,
  dataSuppliers: dataSuppliersState,
  flavors: flavorsState,
  roles: rolesState,
  toast: toastState,
  users: usersState
};

export default combineReducers({
  application,
  dashboard,
  dataSuppliers,
  flavors,
  roles,
  toast,
  users
});
