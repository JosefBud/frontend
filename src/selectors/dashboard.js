import { createSelector } from 'reselect';

export const getDashboard = state => state.dashboard;

export const getDashboardComponent = createSelector(
  getDashboard,
  dashboard => dashboard.dashboard
);

export const getMigrations = createSelector(
  getDashboard,
  dashboard => dashboard.migrations
);
