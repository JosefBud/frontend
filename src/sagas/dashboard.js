// import dayjs from 'dayjs';
// import nanoid from 'nanoid';
import {
  all,
  call,
  put,
  takeLatest
  // delay,
  // take,
  // takeEvery,
  // select
} from 'redux-saga/effects';

import request from 'utils/request';
import { actions, types } from 'reducers/dashboard';

// snake_cased variables here come from RFC 6749
/* eslint-disable camelcase */
function* requestMigrationsWorker() {
  try {
    const endpoint = {
      url: '/data/version',
      method: 'GET'
    };
    const result = yield call(request.execute, { endpoint });

    // update migrations in state or throw an error
    if (result.success) {
      const {
        response: { data }
      } = result;

      yield put(actions.requestMigrationsSuccess(data));
    } else if (result.error) {
      throw result.error;
    } else {
      throw new Error('Failed to get migrations!');
    }
  } catch (error) {
    const { message } = error;

    yield put(actions.requestMigrationsFailure(error));
    yield put(
      actions.popToast({
        title: 'Error',
        icon: 'times-circle',
        message
      })
    );
  }
}

function* requestMigrationsWatcher() {
  yield takeLatest(types.REQUEST_MIGRATIONS, requestMigrationsWorker);
}

export const workers = {
  requestMigrationsWorker
};

export const watchers = {
  requestMigrationsWatcher
};

export default function* saga() {
  yield all(Object.values(watchers).map(watcher => watcher()));
}
