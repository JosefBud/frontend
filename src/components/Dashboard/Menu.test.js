import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { initialState } from 'reducers/dashboard';
import ConnectedMenu from './Menu';
import { withMemoryRouter } from 'utils';

describe('Dashboard <Menu />', () => {
  const mockStore = configureStore();
  const store = mockStore({ dashboard: initialState });
  const RoutedMenu = withMemoryRouter(ConnectedMenu);

  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <RoutedMenu />
      </Provider>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });
});
