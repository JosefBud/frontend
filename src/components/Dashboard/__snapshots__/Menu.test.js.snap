// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard <Menu /> renders correctly 1`] = `
<div
  className="accordion"
>
  <div
    className="card"
  >
    <div
      className="card-body"
    >
      <i
        className="fa [object Object]"
      />
        
      <a
        href="#home"
        onClick={[Function]}
      >
        Dashboard
      </a>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
      onClick={[Function]}
    >
      <h3>
        <i
          className="fa [object Object]"
        />
         Roles
      </h3>
    </div>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="card-body"
      >
        <div>
          <i
            className="fa [object Object]"
          />
            
          <a
            href="#roles"
            onClick={[Function]}
          >
            Roles
          </a>
        </div>
        <div>
          <i
            className="fa [object Object]"
          />
            
          <a
            href="#roles/add"
            onClick={[Function]}
          >
            Add Role
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
      onClick={[Function]}
    >
      <h3>
        <i
          className="fa [object Object]"
        />
         Users
      </h3>
    </div>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="card-body"
      >
        <div>
          <i
            className="fa [object Object]"
          />
            
          <a
            href="#users"
            onClick={[Function]}
          >
            Users
          </a>
        </div>
        <div>
          <i
            className="fa [object Object]"
          />
            
          <a
            href="#users/add"
            onClick={[Function]}
          >
            Add User
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
      onClick={[Function]}
    >
      <h3>
        <i
          className="fa [object Object]"
        />
         Ingredients
      </h3>
    </div>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="card-body"
      >
        <div>
          <i
            className="fa [object Object]"
          />
            
          <a
            href="#flavors"
            onClick={[Function]}
          >
            Flavors
          </a>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card"
  >
    <div
      className="card-header"
      onClick={[Function]}
    >
      <h3>
        <i
          className="fa [object Object]"
        />
         Database
      </h3>
    </div>
    <div
      aria-expanded={null}
      className="collapse"
    >
      <div
        className="card-body"
      >
        <i
          className="fa [object Object]"
        />
          
        <a
          href="#migrations"
          onClick={[Function]}
        >
          Migration History
        </a>
      </div>
    </div>
  </div>
</div>
`;
